apps/api/src/server/applications/search/case-search.routes.js

```diff
@@ -0,0 +1,50 @@
+import express from 'express';
+import { asyncHandler } from '../../middleware/async-handler.js';
+import { getApplicationsByCriteria } from './case-search.controller.js';
+import { validateSearchCriteria} from './search.validators.js';
+
+const router = new express.Router();
+
+router.post(
+	'/',
+	/*
+        #swagger.tags = ['Applications']
+        #swagger.path = '/applications/search'
+        #swagger.description = 'Gets all applications as per search criteria'
+		#swagger.parameters['query'] = {
+            in: 'body',
+			description: 'search criteria searched in Application Reference, Title, or Description',
+			example: 'EN01',
+            type: 'string',
+            required: true
+        }
+		#swagger.parameters['role'] = {
+            in: 'body',
+			description: 'a valid role',
+            type: 'string',
+            required: true
+        }
```

Comment:
we should be able to use an enum here to list which roles are allowed to call this endpoint

Reply:
After a bit of research, it seems like the way of using a sort of enum in JS would be by freezing the object, like adding Object.freeze(validRoles) after its declaration